File: swagger/models/get_products_request.py
from .base_model import BaseModel


class GetProductsRequest(BaseModel):
	"""
	no description available

	Attributes:

		product_ids
		An ID filter for products.

		search_text
		A search filter, used for searching by term.

		category_ids
		A list of category IDs to filter by.

		sub_category_ids
		A list of subcategory IDs to filter by.

		sell_online
		When `true`, only products that can be sold online are returned.<br />
		When `false`, all products are returned.<br />  Default: **false**

		location_id
		The location ID to use to determine the tax for the products that this
		request returns.<br />  Default: **online store**

		limit
		Number of results to include, defaults to 100

		offset
		Page offset, defaults to 0.
	"""

	input_map = {
		'ProductIds': 'product_ids',
		'SearchText': 'search_text',
		'CategoryIds': 'category_ids',
		'SubCategoryIds': 'sub_category_ids',
		'SellOnline': 'sell_online',
		'LocationId': 'location_id',
		'Limit': 'limit',
		'Offset': 'offset',
		}

	output_map = {
		'product_ids': 'ProductIds',
		'search_text': 'SearchText',
		'category_ids': 'CategoryIds',
		'sub_category_ids': 'SubCategoryIds',
		'sell_online': 'SellOnline',
		'location_id': 'LocationId',
		'limit': 'Limit',
		'offset': 'Offset',
		}

	product_ids = None
	search_text = None
	category_ids = None
	sub_category_ids = None
	sell_online = None
	location_id = None
	limit = None
	offset = None

File: swagger/Models/GetProductsRequest.php
<?php

namespace App\MindBody\Models;

/**
 * no description available
 */
class GetProductsRequest extends BaseModel
{
	/**
	 * An ID filter for products.
	 * 
	 * @var string[]
	 */
	public $productIds;
	/**
	 * A search filter, used for searching by term.
	 * 
	 * @var string
	 */
	public $searchText;
	/**
	 * A list of category IDs to filter by.
	 * 
	 * @var integer[]
	 */
	public $categoryIds;
	/**
	 * A list of subcategory IDs to filter by.
	 * 
	 * @var integer[]
	 */
	public $subCategoryIds;
	/**
	 * When `true`, only products that can be sold online are returned.<br />
	 * When `false`, all products are returned.<br />  Default: **false**
	 * 
	 * @var boolean
	 */
	public $sellOnline;
	/**
	 * The location ID to use to determine the tax for the products that this
	 * request returns.<br />  Default: **online store**
	 * 
	 * @var integer
	 */
	public $locationId;
	/**
	 * Number of results to include, defaults to 100
	 * 
	 * @var integer
	 */
	public $limit;
	/**
	 * Page offset, defaults to 0.
	 * 
	 * @var integer
	 */
	public $offset;

	protected function getInputMap()
	{
		return [
			'ProductIds' => 'productIds',
			'SearchText' => 'searchText',
			'CategoryIds' => 'categoryIds',
			'SubCategoryIds' => 'subCategoryIds',
			'SellOnline' => 'sellOnline',
			'LocationId' => 'locationId',
			'Limit' => 'limit',
			'Offset' => 'offset',
			];
	}

	protected function getOutputMap()
	{
		return [
			'productIds' => 'ProductIds',
			'searchText' => 'SearchText',
			'categoryIds' => 'CategoryIds',
			'subCategoryIds' => 'SubCategoryIds',
			'sellOnline' => 'SellOnline',
			'locationId' => 'LocationId',
			'limit' => 'Limit',
			'offset' => 'Offset',
			];
	}
}
